geoSegment.js

```diff
@@ -0,0 +1,353 @@
+/* eslint-disable no-console */
+const fsPromises = require('fs').promises;
+const fs = require('fs-extra');
+const geolib = require('geolib');
+const Joi = require('joi');
+const moment = require('moment-timezone');
+
+const { generateDatesWithinTimeWindow } = require('./utils/generateDatesWithinTimeWindow');
+const { getFirehosePage } = require('./utils/getFirehosePage');
+const { wait } = require('./utils/wait');
+const { loggableAxiosError } = require('./utils/loggableAxiosError');
+const { emptyOrMakeDirectory } = require('./utils/emptyOrMakeDirectory');
+const { Http404Error } = require('./utils/errors');
+
+// File paths
+const CONFIG_FILE_PATH = './config.json';
+const SESSION_SERIES_DIRECTORY_PATH = './sessionseries';
+const OUTPUT_DIRECTORY_PATH = './output';
+
+// Parameters controlling retry backoff
+const RETRY_BACKOFF_MIN = 1;
+const RETRY_BACKOFF_MAX = 1024; // ~ 17 mins
+const RETRY_BACKOFF_EXPONENTIATION_RATE = 2;
+const MIN_REQUEST_DELAY_SECONDS = 0.1;
+
+const SESSION_SERIES_FILE_PATHS = [];
```

Comment:
this is just pedantic but it looks like this isn't a global constant, so shouldn't be upper cased